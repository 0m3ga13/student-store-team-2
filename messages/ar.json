{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
