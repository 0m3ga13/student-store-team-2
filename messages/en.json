{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
