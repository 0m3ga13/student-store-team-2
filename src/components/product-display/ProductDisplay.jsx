"use client";
import React from "react";
import { useRef } from "react";
import { useState, useEffect } from "react";
import Image from "next/image";

/*
{
    id: 2,
    title: "shoes",
    images: [""]
}
*/
// Placeholder object to be replaced with data from Firestore
const imagesArray = [
    "/img/product4.jpg",
    "/img/product3.jpg",
    "/img/product2.jpg",
    "/img/product1.jpg",
];
function ProductDisplay() {
    const [currentImage, setCurrentImage] = useState(imagesArray[0]);
    const indexRef = useRef(1);
    const scrollRef = useRef(null);

    useEffect(() => {
        setInterval(() => {
            setCurrentImage(imagesArray[indexRef.current]);
            scrollRef.current.scrollTo(0, indexRef.current * 249);
            indexRef.current = indexRef.current + 1;
            if (indexRef.current === 4) {
                indexRef.current = 0;
            }
        }, 4000);
    }, []);

    return (
        <>
            <div className=' flex gap-2 w-full'>
                <div
                    ref={scrollRef}
                    className=' h-[560px] scroll-smooth flex flex-col gap-1 overflow-hidden '
                >
                    {imagesArray.map((item, index) => (
                        <button
                            key={index}
                            className='flex flex-col items-center justify-center'
                            onClick={() => setCurrentImage(item)}
                        >
                            <Image
                                src={item}
                                alt='Product'
                                width={166}
                                height={249}
                                className='h-[249px] object-cover rounded'
                            />
                        </button>
                    ))}
                </div>

                <div className='rounded'>
                    <Image
                        src={currentImage}
                        width={500}
                        height={560}
                        alt='Skin Care Products'
                        className='object-cover h-[560px] rounded'
                    />
                </div>
            </div>
        </>
    );
}

export default ProductDisplay;
