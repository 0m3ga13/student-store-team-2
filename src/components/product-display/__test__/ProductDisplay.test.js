import renderer from "react-test-renderer";
import { NextIntlClientProvider } from "next-intl";
import messages from "../../../../messages/en.json";
import ProductDisplay from "../ProductDisplay";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <ProductDisplay />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
