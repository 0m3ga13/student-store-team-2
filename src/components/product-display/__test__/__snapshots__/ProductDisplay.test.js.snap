// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" flex gap-2 w-full"
>
  <div
    className=" h-[560px] scroll-smooth flex flex-col gap-4 overflow-hidden   "
  >
    <button
      className="flex flex-col items-center justify-center"
      onClick={[Function]}
    >
      <img
        alt="Product"
        className="h-[249px] object-cover"
        data-nimg="1"
        decoding="async"
        height={249}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg%2Fproduct4.jpg&w=384&q=75"
        srcSet="/_next/image?url=%2Fimg%2Fproduct4.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproduct4.jpg&w=384&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={166}
      />
    </button>
    <button
      className="flex flex-col items-center justify-center"
      onClick={[Function]}
    >
      <img
        alt="Product"
        className="h-[249px] object-cover"
        data-nimg="1"
        decoding="async"
        height={249}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg%2Fproduct3.jpg&w=384&q=75"
        srcSet="/_next/image?url=%2Fimg%2Fproduct3.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproduct3.jpg&w=384&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={166}
      />
    </button>
    <button
      className="flex flex-col items-center justify-center"
      onClick={[Function]}
    >
      <img
        alt="Product"
        className="h-[249px] object-cover"
        data-nimg="1"
        decoding="async"
        height={249}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg%2Fproduct2.jpg&w=384&q=75"
        srcSet="/_next/image?url=%2Fimg%2Fproduct2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproduct2.jpg&w=384&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={166}
      />
    </button>
    <button
      className="flex flex-col items-center justify-center"
      onClick={[Function]}
    >
      <img
        alt="Product"
        className="h-[249px] object-cover"
        data-nimg="1"
        decoding="async"
        height={249}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg%2Fproduct1.jpg&w=384&q=75"
        srcSet="/_next/image?url=%2Fimg%2Fproduct1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg%2Fproduct1.jpg&w=384&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={166}
      />
    </button>
  </div>
  <div
    className=""
  >
    <img
      alt="Skin Care Products"
      className="object-cover h-[560px] "
      data-nimg="1"
      decoding="async"
      height={560}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg%2Fproduct4.jpg&w=1080&q=75"
      srcSet="/_next/image?url=%2Fimg%2Fproduct4.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fproduct4.jpg&w=1080&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={500}
    />
  </div>
</div>
`;
